src/LaunchDarkly.ServerSdk/Files/FlagFileData.cs

```diff
@@ -17,46 +18,58 @@ class FlagFileData
         [JsonProperty(PropertyName = "segments", NullValueHandling = NullValueHandling.Ignore)]
         public Dictionary<string, JToken> Segments { get; set; }
 
-        public void AddToData(IDictionary<IVersionedDataKind, IDictionary<string, IVersionedData>> allData)
+        public void AddToData(IDictionary<IVersionedDataKind, IDictionary<string, IVersionedData>> allData,
+            DuplicateKeysHandling duplicateKeysHandling)

```

Comment:
How about making `duplicateKeysHandling` a field that's set in the constructor, so you don't have to keep passing it to `AddItem`?

Reply:
The relationship between `FlagFileData` and `FlagFileParser` makes this awkward.  I pulled out the `AddToData` method into its own class in a alternate PR:

https://github.com/launchdarkly/dotnet-server-sdk/pull/123/files

Reply:
Oh yeah, I see what you mean. Sorry, I wrote this code but it's been a while.

At this point though, I'm finding it slightly hard to follow three nearly identical PRs for the same feature (especially since Github does not make it easy to see at a glance what the differences are between them). Could you please not open any more PRs for this in the near future, and just pick one to focus on— if you have any more proposed changes, just make them on the same branch and then if they're not desirable they can easily be reverted. I think the newest one, 123, is what I would prefer.